daemon/graphdriver/vfs/driver.go

```diff
@@ -69,11 +81,32 @@ func (d *Driver) Cleanup() error {
 	return nil
 }
 
+func (d *Driver) parseOptions(options []string) error {
+	for _, option := range options {
+		key, val, err := parsers.ParseKeyValueOpt(option)
+		if err != nil {
+			return errdefs.InvalidParameter(err)
+		}
+		switch key {
+		case "size":
+			size, err := units.RAMInBytes(val)
+			if err != nil {
+				return errdefs.InvalidParameter(err)
+			}
+			if err = d.setQuotaOpt(uint64(size)); err != nil {
+				return errdefs.InvalidParameter(errors.Errorf("failed to set option size for vfs: %v", err))
+			}
+		default:
+			return errdefs.InvalidParameter(errors.Errorf("unknown option %s for vfs", key))
+		}
+	}
+	return nil
+}
+
 // CreateReadWrite creates a layer that is writable for use as a container
 // file system.
 func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
-	var err error
-	var size int64
+	quotaSize := d.getQuotaOpt()
 
 	if opts != nil {
 		for key, val := range opts.StorageOpt {
```

Comment:
Isn't this doing the same as `parseOptions`? If so, can we reuse that function here?

Reply:
This parses the parameters of the container, and parseOptions parses the parameters of the docker daemon. Considering scalability, I didn't reuse this function.